src/coreclr/src/jit/optcse.cpp

```diff
@@ -401,10 +401,27 @@ void Compiler::optValnumCSE_Init()
 //
 unsigned Compiler::optValnumCSE_Index(GenTree* tree, Statement* stmt)
 {
-    unsigned key;
+    size_t   key;
     unsigned hash;
     unsigned hval;
     CSEdsc*  hashDsc;
+    bool     isIntConstHash       = false;
+    bool     enableSharedConstCSE = false;
+    int      configValue          = JitConfig.JitConstCSE();
+
+#if defined(TARGET_ARM64)
+    // ARM64 - allow to combine with nearby offsets, when config is not 2 or 4
+    if ((configValue != 2) && (configValue != 4))
```

Comment:
```suggestion
    if (configValue == 0)
```

I think you are doing some extra work that is not needed when `configValue == 1` for ARM64, based on the previous condition it would set `enableSharedConstCSE ` and it would enable block under `else if (enableSharedConstCSE && tree->IsIntegralConst())`

Reply:
There could be additional values added beyond 0,1,2,3,4 with different meanings added in the future.

Reply:
It would be nice to define constants for these values, with comments similar to what you've provided in jitconfigvalues.h - and then it would be easier to reason about what these mean.

Reply:
I will add defines for these constants